### PR TITLE
Added cumulative_logsumexp to Tensorflow frontend

### LINKED ISSUE #14606: cumulative_logsumexp
Implementing `cumulative_logsumexp` for the `TensorFlow` frontend.

### PR DESCRIPTION
**Pull request by roboFiddle**
Closes #14606

### CODE REVIEW

File: ivy/functional/frontends/tensorflow/math.py [hunk 1]
```diff
@@ -144,6 +144,23 @@ def cumsum(x, axis, exclusive=False, reverse=False, name=None):
...
 
 
+@to_ivy_arrays_and_back
+def cumulative_logsumexp(
+    x, axis, exclusive=False, reverse=False, name="cumulative_logsumexp"
+):
+    return ivy.astype(
+        ivy.log(
+            ivy.cumsum(
+                ivy.exp(ivy.astype(x, "float64")),
```
Comment (Dsantra92) [inline ivy/functional/frontends/tensorflow/math.py:154]: Not sure if the type conversion is needed here.

**Review by Dsantra92 — COMMENTED** [review]
Hi @roboFiddle, thanks for the PR. The changes look good overall but I had question on the type conversion happening here.

**Comment by roboFiddle** [comment]
I just refactored the function to avoid the type conversion. It's a little more faithful to the original Tensorflow version, which is implemented using a scan. Let me know if you have any questions about it.

**Comment by roboFiddle** [comment]
@YushaArif99 Would it be able to get this push request reviewed again? If any changes are needed, please let me know.

File: ivy_tests/array_api_testing/test_array_api
Comment (YushaArif99) [inline ivy_tests/array_api_testing/test_array_api:1]: Also, can you kindly remove this from the modified files 🙂

**Comment by roboFiddle** [comment]
I've resolved the test error (the "float" dtypes as a category in Ivy includes some types not valid for cumulative_logsumexp. The merge conflict seems to be that there's a lot of overlapping code between the two test functions - do I need to resolve that myself?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 2]
```diff
@@ -2527,3 +2527,34 @@ def test_tensorflow_in_top_k(
...
+        max_value=10,
+    ),
+    exclusive=st.booleans(),
+    reverse=st.booleans(),
+)
+def test_tensorflow_cumulative_logsumexp(
+    *, dtype_x_axis, frontend, test_flags, fn_tree, on_device, exclusive, reverse
+):
+    input_dtype, x, axis = dtype_x_axis
+    assume(input_dtype != "bfloat16")
```
Comment (YushaArif99) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:2549]: You dont need to add any extra logic to filter out specific `dtypes` here. You can instead use the `@with_unsupported_dtypes` decorator in the `cumulative_logsumexp` function itself to achieve this purpose. You can read more about the use of this decorator [here](https://unify.ai/docs/ivy/overview/deep_dive/data_types.html#supported-and-unsupported-data-types)

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @roboFiddle 

Can you kindly address the abovementioned issue and re-request a review once that is fixed. Also, can you kindly resolve the merge conflicts in this PR so that the CI tests can rerun. Thanks 🙂

**Pull request closed without merging**